<?xml version="1.0"?>
<launch>
  <arg name="goal_x" default="7.225" />
  <arg name="goal_y" default="-6.785" />
  <arg name="goal_theta" default="-1.57079632679" />

  <arg name="no_static_map" default="true"/>

  <arg name="base_global_planner" default="navfn/NavfnROS"/>
  <arg name="base_local_planner" default="dwa_local_planner/DWAPlannerROS"/>
  <!-- <arg name="base_local_planner" default="base_local_planner/TrajectoryPlannerROS"/> -->

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" >

    <param name="base_global_planner" value="$(arg base_global_planner)"/>
    <param name="base_local_planner" value="$(arg base_local_planner)"/>  
    <rosparam file="$(find parc-solution)/config/planner.yaml" command="load"/>

    <!-- observation sources located in costmap_common.yaml -->
    <rosparam file="$(find parc-solution)/config/costmap_common.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find parc-solution)/config/costmap_common.yaml" command="load" ns="local_costmap" />

    <!-- local costmap, needs size -->
    <rosparam file="$(find parc-solution)/config/costmap_local.yaml" command="load" ns="local_costmap" />
    <param name="local_costmap/width" value="3.0"/>
    <param name="local_costmap/height" value="3.0"/>

    <!-- static global costmap, static map provides size -->
    <rosparam file="$(find parc-solution)/config/costmap_global_static.yaml" command="load" ns="global_costmap" unless="$(arg no_static_map)"/>
    
    <!-- global costmap with laser, for odom_navigation_demo -->
    <rosparam file="$(find parc-solution)/config/costmap_global_laser.yaml" command="load" ns="global_costmap" if="$(arg no_static_map)"/>
    <param name="global_costmap/width" value="100.0" if="$(arg no_static_map)"/>
    <param name="global_costmap/height" value="100.0" if="$(arg no_static_map)"/>
  </node>

  <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" />

  <node pkg="nodelet" type="nodelet" name="nodelet1"
        args="load depth_image_proc/point_cloud_xyz nodelet_manager">
    <remap from="camera_info" to="/camera/depth/camera_info"/>
    <remap from="image_rect" to="/camera/depth/image_raw"/>
    <remap from="points" to="/camera/depth/points"/>
  </node>

  <node name="task3_solution" pkg="parc-solution" type="task3_node.py" args="$(arg goal_x) $(arg goal_y) $(arg theta)" output="screen" />

</launch>
